api/firmware/query.go

```diff
@@ -70,27 +71,37 @@ func (device *Device) rawQueryV7(msg []byte) ([]byte, error) {
 		break
 	}
 
-	var lastQueryTime time.Time
-
+	// saving 11 timestamps to show the last 10 intervals.
+	lastQueryTimes := ring.New(11)
```

Comment:
TIL about ring! would be nice if golang added a generic version of it to the stdlib.